rewrite-java-test/src/test/java/org/openrewrite/java/cleanup/ReplaceRedundantFormatWithPrintfTest.java

```diff
@@ -0,0 +1,293 @@
+             
+                  void test(PrintStreamSubclass stream, String arg) {
+                      stream.println(String.format("hello %s", arg));
+                  }
+              }
+              """,
+            """
+              class Test {
+                  static class PrintStreamSubclass extends java.io.PrintStream {}
+                  
+                  void test(PrintStreamSubclass stream, String arg) {
+                      stream.printf("hello %s%n", arg);
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void transformsPrint() {
+        rewriteRun(
+          java(
+            """
+              class Test {
+                  void test(String arg) {
+                      System.out.print(String.format("hello %s", arg));
+                  }
+              }
+              """,
+            """
+              class Test {
+                  void test(String arg) {
+                      System.out.printf("hello %s", arg);
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void transformsPrintWithTextBlockFormatString() {
+        rewriteRun(
+          java(
+            """
+              class Test {
+                  void test(String arg) {
+                      System.out.print(String.format(\"\"\"
+                      hello %s\"\"\", arg));
+                  }
+              }
+              """,
+            """
+              class Test {
+                  void test(String arg) {
+                      System.out.printf(\"\"\"
+                      hello %s\"\"\", arg);
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void appendsNewlineForPrintln() {
+        rewriteRun(
+          java(
+            """
+              class Test {
+                  void test(String arg) {
+                      System.out.println(String.format("hello %s", arg));
+                  }
+              }
+              """,
+            """
+              class Test {
+                  void test(String arg) {
+                      System.out.printf("hello %s%n", arg);
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void appendsNewlineForPrintlnWithTextBlockFormatString() {
+        rewriteRun(
+          java(
+            """
+              class Test {
+                  void test(String arg) {
+                      System.out.println(String.format(\"\"\"
+                      hello %s\"\"\", arg));
+                  }
+              }
+              """,
+            """
+              class Test {
+                  void test(String arg) {
+                      System.out.printf(\"\"\"
+                      hello %s%n\"\"\", arg);
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
```

Comment:
not sure if these tests are gratuitous; I was concerned about how I'd built the `JavaTemplate` code

Reply:
Seems reasonable enough to me